index.js

```diff
@@ -409,46 +420,71 @@ const fileType = input => {
 		};
 	}
 
-	// Check for MPEG header at different starting offsets
+	let flagId3 = false;
+
 	for (let start = 0; start < 2 && start < (buffer.length - 16); start++) {
-		if (
-			check([0x49, 0x44, 0x33], {offset: start}) || // ID3 header
-			check([0xFF, 0xE2], {offset: start, mask: [0xFF, 0xE6]}) // MPEG 1 or 2 Layer 3 header
-		) {
-			return {
-				ext: 'mp3',
-				mime: 'audio/mpeg'
-			};
+		// Check for ID3 header
+		if (buffer.length >= start + 10 && checkString('ID3', {offset: start})) {
+			const id3Len = readUINT32SYNCSAFE(buffer, start + 6);
+			start += (10 + id3Len - 1); // Skip ID3 header
+			flagId3 = true;
+			continue;
 		}
 
-		if (
-			check([0xFF, 0xE4], {offset: start, mask: [0xFF, 0xE6]}) // MPEG 1 or 2 Layer 2 header
-		) {
-			return {
-				ext: 'mp2',
-				mime: 'audio/mpeg'
-			};
-		}
+		// Check MPEG 1 or 2 Layer 3 header, or 'layer 0' for ADTS (MPEG sync-word 0xFFE)
+		if (buffer.length >= start + 2 && check([0xFF, 0xE0], {offset: start, mask: [0xFF, 0xE0]})) {
+			// Check for ADTS header (last bit of sync-word 0xFFF & layer=0)
+			if (check([0x10], {offset: start + 1, mask: [0x16]})) {
+				// Check for (ADTS) MPEG-2
+				if (check([0x08], {offset: start + 1, mask: [0x08]})) {
+					return {
```

Comment:
this returns exactly the same as alternative

Reply:
I mean,
```js
				// Check for (ADTS) MPEG-2
				if (check([0x08], {offset: start + 1, mask: [0x08]})) {
					return {
						ext: 'aac',
						mime: 'audio/aac'
					};
				}

				// Must be (ADTS) MPEG-4
				return {
					ext: 'aac',
					mime: 'audio/aac'
				};
```

Reply:
True, I created the distinction 'between MPEG-2 and MPEG-4.
I updated the return types of MPEG-4 pending a sample file for MPEG-2.

Because I wanted to have some evidence that this was done like that for files like that,

Then I found out such a fixture was already present. I updated the return type, and I thought, maybe good to the distinct to keep as is, just for clarity we covered both cases, and allow to pin point to that specific case if there is still an issue.